api/src/routes/settings.test.ts

```diff
@@ -271,6 +271,76 @@ describe('settingRoutes', () => {
       });
     });
 
+    describe('/update-my-about', () => {
+      test('PUT updates the values in about settings', async () => {
+        const response = await superRequest('/update-my-about', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          about: 'Teacher at freeCodeCamp',
+          name: 'Quincy Larson',
+          location: 'USA',
+          picture:
+            'https://cdn.freecodecamp.org/platform/english/images/quincy-larson-signature.svg'
+        });
+
+        expect(response.body).toEqual({
+          message: 'flash.updated-about-me',
+          type: 'success'
+        });
+
+        const user = await fastifyTestInstance?.prisma.user.findFirst({
+          where: { email: 'foo@bar.com' }
+        });
+
+        expect(user?.about).toEqual('Teacher at freeCodeCamp');
+        expect(user?.name).toEqual('Quincy Larson');
+        expect(user?.location).toEqual('USA');
+        expect(user?.picture).toEqual(
+          'https://cdn.freecodecamp.org/platform/english/images/quincy-larson-signature.svg'
+        );
+        expect(response.statusCode).toEqual(200);
+      });
+
+      test('PUT updates the values in about settings without image', async () => {
+        const response = await superRequest('/update-my-about', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          about: 'Teacher at freeCodeCamp',
+          name: 'Quincy Larson',
+          location: 'USA',
+          // `new URL` throws if the image isn't a URL, this checks if it doesn't throw.
+          picture: ''
+        });
+
+        expect(response.body).toEqual({
+          message: 'flash.updated-about-me',
+          type: 'success'
+        });
+        expect(response.statusCode).toEqual(200);
+      });
+
+      test('PUT throw 500 if the image is invalid', async () => {
+        const response = await superRequest('/update-my-about', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          about: 'Teacher at freeCodeCamp',
+          name: 'Quincy Larson',
+          location: 'USA',
+          picture: 'invaild'
+        });
+
+        expect(response.body).toEqual({
+          message:
+            'Failed to serialize an error. Error: "type" is required!. Original error: cb is not a function',
+          code: 'FST_ERR_FAILED_ERROR_SERIALIZATION',
+          statusCode: 500
+        });
```

Comment:
We should never see these errors in practice, since it means that there's an error in our code. What should happen is that the response we send is valid according to the response schema.

So, in this case we should get a success response, because the current api simply ignores strings that are not urls.

Reply:
I am wondering if that is possible to mimic in the new API, and how?

Reply:
The new API can send whatever response we want, but it's returning this because there's an uncaught error thrown while it tries to handle the request. I'll add a suggestion to fix it, below.